### PR TITLE
openmpi: disable openmpi-devel-* subports

### PR DESCRIPTION
**Pull request by mascguy**
#### Description

Per ticket from @jmroot, disable openmpi-devel-* subports.

Fixes: [openmpi-devel-default: The requested URL returned error: 403 Forbidden](https://trac.macports.org/ticket/62068)

###### Type(s)
- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 10.8.5 12F2560
Xcode 5.1.1 5B1008

macOS 10.9.5 13F1911
Xcode 6.2 6C131e

macOS 10.10.5 14F2511
Xcode 6.4 6E35b

macOS 10.11.6 15G22010
Xcode 7.3.1 7D1014

macOS 10.12.6 16G2136
Xcode 9.2 9C40b

macOS 10.13.6 17G14019
Xcode 10.1 10B61

macOS 10.14.6 18G103
Xcode 11.3.1 11C505

macOS 10.15.6 19G2021
Xcode 12.0 12A7209

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by mascguy** [comment]
@jmroot Are changes to portgroup `mpi` necessary, or are these changes sufficient?

**Comment by mascguy** [comment]
This looks good to me. @jmroot, any final thoughts, before we commit this?

**Comment by jmroot** [comment]
> @jmroot Are updates to portgroup `mpi` necessary, or are these changes sufficient?

Yes, the openmpi_devel entries need to be removed from the mpidb array.

**Review by jmroot — COMMENTED** [review]
Doing it this way will leave the -devel subports installed for anyone who already installed them. An alternative would be to include the obsolete portgroup and mark each one as `replaced_by` its non-devel counterpart, which would cause users to be automatically upgraded from the old snapshot to the current release version.

**Comment by mascguy** [comment]
> Doing it this way will leave the -devel subports installed for anyone who already installed them. An alternative would be to include the obsolete portgroup and mark each one as `replaced_by` its non-devel counterpart, which would cause users to be automatically upgraded from the old snapshot to the current release version.

Great idea! I'll make the changes, and report back.

**Comment by mascguy** [comment]
> Yes, the openmpi_devel entries need to be removed from the mpidb array.

Just to confirm... we still want to make this change, with the `replaced_by` strategy?

**Comment by mascguy** [comment]
Changes pushed.

**Comment by jmroot** [comment]
> > Yes, the openmpi_devel entries need to be removed from the mpidb array.
> 
> Just to confirm... we still want to make this change, with the `replaced_by` strategy?

Yes, you don't want variants depending on replaced ports.

**Comment by mascguy** [comment]
> > Just to confirm... we still want to make this change, with the `replaced_by` strategy?
> 
> Yes, you don't want variants depending on replaced ports.

Sounds good, all set.

File: science/openmpi/Portfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ PortGroup           select 1.0
 PortGroup           compilers 1.0
 PortGroup           muniversal 1.0
 PortGroup           legacysupport 1.0
+PortGroup           obsolete 1.0
```
Comment (jmroot) [inline science/openmpi/Portfile:9]: This makes all the subports obsolete. Move this to around line 49, inside the `if {[string first "-devel" $subport] > 0} {` block.

Comment (mascguy) [reply science/openmpi/Portfile]: Done

**Review by jmroot — APPROVED** [review]
LGTM

**Comment by mascguy** [comment]
Thanks Josh!

**Pull request merged**